src/device-registry/utils/create-device.js

```diff
@@ -749,12 +749,15 @@ const createDevice = {
       const baseURL = constants.CREATE_THING_URL;
       const { body } = request;
       const { category } = body;
-      const data = body;
+      let data = body;
+      if (isEmpty(data.long_name) && !isEmpty(data.name)) {
+        data.long_name = data.name;
+      }
       const map = constants.DEVICE_THINGSPEAK_MAPPINGS;
       let context = {};
       if (category === "bam") {
         context = constants.BAM_THINGSPEAK_FIELD_DESCRIPTIONS;
-      } else if (category === "lowcost") {
+      } else {
         context = constants.THINGSPEAK_FIELD_DESCRIPTIONS;
```

Comment:
for cases where device `category` is not provided, just use the "lowcost" TS configuration.